pyproject.toml

```diff
@@ -1,3 +1,47 @@
+[project]
+name = "spidev"
+description = "Python bindings for Linux SPI access through spidev"
+dynamic = ["version", "readme"]
+
+authors = [
+    { name = "Volker Thoms", email = "unconnected@gmx.de" },
+]
+maintainers = [
+    { name = "Stephen Caudle", email = "scaudle@doceme.com" },
+]
+license = { text = "MIT", file = "LICENSE" }
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Operating System :: POSIX :: Linux",
+    "License :: OSI Approved :: MIT License",
+    "Intended Audience :: Developers",
+    "Programming Language :: Python :: 2.6",
+    "Programming Language :: Python :: 2.7",
+    "Programming Language :: Python :: 3",
+    "Topic :: Software Development",
+    "Topic :: System :: Hardware",
+    "Topic :: System :: Hardware :: Hardware Drivers",
+]
+keywords = ["SPI", "Linux", "SPI device"]
+requires-python = ">=3.8"
+
 [build-system]
 requires = ["setuptools>=61.0"]
 build-backend = "setuptools.build_meta"
+
+[project.urls]
+Repository = "http://github.com/doceme/py-spidev"
+Issues = "https://github.com/doceme/py-spidev/issues"
+Changelog = "https://github.com/doceme/py-spidev/blob/master/CHANGELOG.md"
+
+[tool.setuptools]
+ext-modules = [
+    { name = "spidev", sources = ["spidev_module.c"] }
+]
+
+[tool.setuptools.package-data]
+spidev = ["spidev_module.c", "README.md", "CHANGELOG.md", "LICENSE"]
+
+[tool.setuptools.dynamic]
+version = { file = "spidev_module.c", pattern = "#define.*_VERSION_.*\"([^\"]+)\"" }
```

Comment:
What version of `setuptools` are you using? I don't find any documentation of the `pattern` property. We probably need to keep the code in `setup.py` to set the dynamic version. At least for now, as long as the version is defined in the C file.